demos/rag_agentic/notebooks/level4_agentic_and_mcp.ipynb

```diff
@@ -0,0 +1,129 @@
+      "metadata": {}
+    },
+    {
+      "id": "4481ba68",
+      "cell_type": "code",
+      "source": "!pip install llama-stack-client==0.1.9 llama-stack==0.1.9",
+      "metadata": {},
+      "outputs": [],
+      "execution_count": null
+    },
+    {
+      "id": "25fc0a44",
+      "cell_type": "code",
+      "source": "from llama_stack_client.lib.agents.event_logger import EventLogger\nimport logging\n\nlogger = logging.getLogger(__name__)\nlogger.setLevel(logging.INFO)\nstream_handler = logging.StreamHandler()\nstream_handler.setLevel(logging.INFO)\nformatter = logging.Formatter('%(message)s')\nstream_handler.setFormatter(formatter)\nlogger.addHandler(stream_handler)",
+      "metadata": {},
+      "outputs": [],
+      "execution_count": 3
+    },
+    {
+      "id": "681412e1",
+      "cell_type": "markdown",
+      "source": "### Connecting to llama-stack server\n\nFor the llama-stack instance, you can either run it locally or connect to a remote llama-stack instance.\n\n#### Remote llama-stack\n\n- For remote, be sure to set `remote` to `True` and populate the `remote_llama_stack_endpoint` variable with your llama-stack remote.\n- [Remote Setup Guide](https://github.com/opendatahub-io/llama-stack-on-ocp/tree/main/kubernetes)\n\n#### Local llama-stack\n- For local, be sure to set `remote` to `False` and validate the `local_llama_stack_endpoint` variable. It is based off of the default llama-stack port which is `8321` but is configurable with your deployment of llama-stack.\n- [Local Setup Guide](https://github.com/redhat-et/agent-frameworks/tree/main/prototype/frameworks/llamastack)",
+      "metadata": {}
+    },
+    {
+      "id": "6fa38ad0",
+      "cell_type": "code",
+      "source": "remote = True # Use the `remote` variable to switching between a local development environment and a remote kubernetes cluster.\nmodel=\"meta-llama/Llama-3.2-3B-Instruct\"\n\nremote_llama_stack_endpoint = \"your-llama-stack-endpoint\"  # Replace with your llama-stack endpoint if remote is set to True\nlocal_llama_stack_endpoint = \"http://localhost:8321\"\n\ntavily_search_api_key = \"tavily-api-key\" # Replace with your Tavily API key (required for demo 2)\n\nfrom llama_stack_client import LlamaStackClient\n\nif remote:\n    base_url = remote_llama_stack_endpoint\nelse:\n    base_url = local_llama_stack_endpoint\n\nclient = LlamaStackClient(\n    base_url=base_url,\n    provider_data={\n        \"tavily_search_api_key\": tavily_search_api_key # This is required for demo 2\n    })\nlogger.info(f\"Connected to Llama Stack server @ {base_url} \\n\")",
```

Comment:
Can we keep using env variables here instead of adding creds and endpoints to the notebook?